BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -32,6 +33,63 @@ public class RealSense2 : Camera, IDisposable
         private bool _pipelineRunning = false;
         private float _depthScale = 1.0f;
 
+        #region Filter
+        public class RealSense2Filter
+        {
+            private ProcessingBlock _filter;
+
+            public bool Enabled { get; set; }
+            public Sensor.SensorOptions Options { get => _filter.Options; }
+
+            public RealSense2Filter(ProcessingBlock filter)
+            {
+                _filter = filter;
+            }
+
+            internal VideoFrame ApplyFilter(VideoFrame frame)
+            {
+                if (_filter is DecimationFilter)
```

Comment:
Candy: Could be written as `if (_filter is DecimationFilter df)`, and then you can use `df` inside the if-block without need for the additional `as DecimationFilter`.